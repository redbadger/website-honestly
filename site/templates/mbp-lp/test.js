import React from 'react';
import { render, shallow } from 'enzyme';
import MBPLP from '.';

describe('site/templates/gold-coin-lp', () => {
  const testMBPPage = {
    slug: 'test',
    title: 'test-title',
    time: 'string',
    date: 'string',
    place: 'string',
    location: 'string',
    consultants: [
      {
        image: 'https://www.example.com/test-3.jpg',
        name: 'test-consultant',
        role: 'test-role',
        profileUrl: 'https://www.example.com/test-consultant',
      },
    ],
    formId: 'abc123',
    hubspotForm: {
      portalId: 1111111,
      guid: 'test-guid',
      name: 'test-name',
      cssClass: 'test-css',
      submitText: 'test-submit',
      inlineMessage: 'test-inline',
      formFields: [
        {
          name: 'test-field-name',
          label: 'test-label',
          fieldType: 'lunch',
          required: true,
          enabled: true,
          hidden: false,
          labelHidden: false,
        },
      ],
    },
  };

  it('Renders', () => {
    const shallowRender = props => shallow(<MBPLP {...props} />);
    const component = shallowRender({ ...testMBPPage });
    expect(component).toMatchSnapshot();
  });

  it('should render consultants', () => {
    const mbpPage = render(<MBPLP {...testMBPPage} />);

    const consultantList = mbpPage.find('.contentSpeakers').last();
    expect(consultantList.length).toEqual(1);
    expect(consultantList.text()).toMatch(/test-consultant/);
  });
});
