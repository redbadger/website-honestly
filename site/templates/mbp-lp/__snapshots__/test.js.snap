// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/templates/gold-coin-lp Renders 1`] = `
<div>
  <Social
    altText="Mission Beyond Product Logo"
    description="Mission Beyond Product is an event series enabling collective cooperation, where thinkers and doers come together to tackle the world’s biggest challenges."
    metaImage="mbp-meta-logo.png"
    title="Mission Beyond Product"
    url="https://missionbeyond.co.uk"
  />
  <div
    className="MBPLP"
  >
    <div
      className="mainContent"
    >
      <div
        className="header"
      >
        <div
          className="headerLogo"
        >
          <MBPLogo />
        </div>
      </div>
      <div
        className="contentHeaderText narrowContent"
      >
        <h3>
          What?
        </h3>
        <p>
          Mission Beyond Product is a coalition of action-taking business leaders ready to step beyond the walls of their own organisations. Together, we’ll tackle the world’s biggest challenges by creating bold new ideas and transforming words into action.
        </p>
        <a
          className="headerAnchor"
          href="#mbp-form"
        >
          RSVP
        </a>
      </div>
      <div
        className="triangleContainer"
      >
        <TriangleGraphic />
      </div>
      <div
        className="contentText contentTextLarge"
      >
        <p>
          Join us at the first event online with 
          <strong>
            Matthew Syed
          </strong>
          , a bestselling author and 
          <strong>
            Cain Ullah
          </strong>
          , CEO of Red Badger for a breakfast masterclass in creative problem-solving for the future. As part of the event, we will demo ShareThyme, a platform that connects generations through cooking, and invite you to be part of our next mission.
        </p>
      </div>
      <div
        className="columnContainer narrowContent"
      >
        <div
          className="contentTextColumn"
        >
          <div
            className="contentText"
          >
            <div
              className="anchorCushion"
              id="mbp-form"
            />
            <h2>
              Event details
            </h2>
            <p>
              <strong>
                Date:
              </strong>
               
              string
              <br />
              <strong>
                Time:
              </strong>
               
              string
              <br />
              <strong>
                Place:
              </strong>
               
              string
              <br />
              <strong>
                Location:
              </strong>
               
              string
            </p>
            <p>
              <strong>
                This is an invitation-only event.
              </strong>
            </p>
          </div>
          <Agenda
            className="displayLarge"
          />
        </div>
        <div
          className="MBPLPForm"
        >
          <div>
            <HubspotForm
              consentCssClass="MBPLPFormConsent"
              cssClass="test-css goldCoinForm"
              formFields={
                Array [
                  Object {
                    "enabled": true,
                    "fieldType": "lunch",
                    "hidden": false,
                    "label": "test-label",
                    "labelHidden": false,
                    "name": "test-field-name",
                    "required": true,
                  },
                ]
              }
              guid="test-guid"
              inlineMessage="test-inline"
              name="test-name"
              pageTitle="test-title"
              portalId={1111111}
              submitText="test-submit"
            />
          </div>
        </div>
        <Agenda
          className="displaySmall"
        />
      </div>
      <div
        className="narrowContent"
      >
        <div
          className="contentText"
        >
          <h2
            className="speakers"
          >
            Speakers
          </h2>
          <div
            className="contentSpeakers"
          >
            <Consultant
              image="https://www.example.com/test-3.jpg"
              key="test-consultant"
              name="test-consultant"
              profileUrl="https://www.example.com/test-consultant"
            />
          </div>
        </div>
        <div
          className="contentText contentTextAOC"
        >
          <h2>
            Questions?
          </h2>
          If you have any queries, please get in touch with us
           
          <a
            className="inlineLink"
            href="mailto:hello@missionbeyond.co.uk"
          >
            here.
          </a>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
